### ISSUE TITLE
[Feature Request] [Training] Support Fused Adam for CPU

### ISSUE THREAD

**Issue by Epliz** [microsoft/onnxruntime#17403]
### Describe the feature request

Hi,
It looks like Fused Adam is not supported on CPU at the moment.
I get the following error whenever trying to use it on CPU:
```
  File "/home/me/git/ml/venv/lib/python3.10/site-packages/onnxruntime/training/optim/fused_adam.py", line 86, in __init__
    self._dummy_overflow_buf = torch.cuda.IntTensor([0])
TypeError: type torch.cuda.IntTensor not available. Torch not compiled with CUDA enabled.
```
And a clear "Currently GPU-only." when looking at the code.

Can you please add that support?

### Describe scenario use case

Trying to use ONNX RT Training + Torch to train a model using AdamW.

**Comment by pengwa** [comment]
Are you looking for further performance gains with a fused adam in CPU? If in this case, I would doubt whether it will bring such a benefit. The fusion here on CUDA tried to launch less kernels with each do multiple param's update, to avoid too many small CUDA kernel launches. Add @baijumeswani as FYI.

**Comment by Epliz** [comment]
I think it could have performance gains for sure in some situations as the fusion saves memory bandwidth and CPUs tend to easily get memory bound. It's hard to tell without having the optimization implemented.

But I would also just appreciate if the same code worked both on GPUs and CPUs without changing any code.

**Comment by baijumeswani** [comment]
`FusedAdam` uses the multi tensor apply so we avoid having to loop over all the model parameters and can work on them in parallel. Unfortunately, we don't have an implementation right now to do that on cpu.

`FusedAdam` numerically should be equivalent to either [transformers.AdamW](https://huggingface.co/docs/transformers/main/main_classes/optimizer_schedules#transformers.AdamW) if `adam_w_mode` is `ADAMW_TRANSFORMERS` or [torch.AdamW](https://pytorch.org/docs/stable/generated/torch.optim.AdamW.html#torch.optim.AdamW) if the `adam_w_mode` is `ADAMW_TORCH`.

We typically expect users to use cuda for large model training and for scenarios where perf is important which is also the reason that we do not have a stable cpu release for onnxruntime-training. But I do see your point of having code work for both cpu and cuda.

@Epliz would you be open to the idea of adding a fallback to transformers.AdamW or torch.AdamW in case user selects cpu as their device?

**Comment by Epliz** [comment]
Yes, that would be fine by me to just have a fallback and maybe log a warning once to indicate to users that it is using the fallback
